lib/parse.js

```diff
@@ -493,9 +492,31 @@ var MERGEABLE_NODES = {};
  * @return {Object} - `prev`.
  */
 MERGEABLE_NODES.text = function (prev, node) {
-    prev.value += node.value;
+    if (unmergeable(prev) || unmergeable(node)) {
+        return node;
+    } else {
+        prev.value += node.value;
+        return prev;
+    }
 
-    return prev;
+    /**
+     * Check whether node is mergeable with adjacent nodes.
+     *
+     * @param {Object} node
+     * @return {boolean} - False if mergeable, true if not.
+     */
+    function unmergeable(node) {
+        if (!node.position) {
+            return true;
+        }
+
+        var start = node.position.start;
+        var end = node.position.end;
+
+        // Nodes with such a position info cannot be merged.
+        return start.line == end.line &&
+            end.column - start.column > node.value.length;
+    }
```

Comment:
Right, here you’re checking if these nodes represent a larger space than the actual size of their content right? Nice work!

Could you hoist it into the top-most scope? And, place it before their first use (I prefer my declarations before the spots they’re used :wink:)